packages/gatsby/src/commands/develop-process.ts

```diff
@@ -108,34 +128,114 @@ module.exports = async (program: IProgram): Promise<void> => {
+
+            // These were previously in `bootstrap()` but are now
+            // in part of the state machine that hasn't been added yet
+            await postBootstrap({ parentSpan: bootstrapSpan })
+            await rebuildSchemaWithSitePage({ parentSpan: bootstrapSpan })
+
+            await extractQueries({ parentSpan: bootstrapSpan })
+            await writeOutRedirects({ parentSpan: bootstrapSpan })
+
+            startRedirectListener()
+            bootstrapSpan.finish()
+
+            // These are the parts that weren't in bootstrap
+
+            // Start the createPages hot reloader.
+            bootstrapPageHotReloader(gatsbyNodeGraphQLFunction)
+
+            // Start the schema hot reloader.
+            bootstrapSchemaHotReloader()
+
+            const { queryIds } = await calculateDirtyQueries({ store })
+
+            await runStaticQueries({ queryIds, store, program, graphqlRunner })
+            await runPageQueries({ queryIds, store, program, graphqlRunner })
+            await writeOutRequires({ store })
+            boundActionCreators.setProgramStatus(
+              ProgramStatus.BOOTSTRAP_QUERY_RUNNING_FINISHED
+            )
+
+            await db.saveState()
+
+            await waitUntilAllJobsComplete()
+            requiresWriter.startListener()
+            db.startAutosave()
+            queryUtil.startListeningToDevelopQueue({
+              graphqlTracing: program.graphqlTracing,
+            })
+            queryWatcher.startWatchDeletePage()
+
+            await startWebpackServer({ program, app, workerPool })
+          },
+        },
+      },
+    },
+  }
 
-  await runStaticQueries({ queryIds, store, program })
-  await runPageQueries({ queryIds, store, program })
-  await writeOutRequires({ store })
-  boundActionCreators.setProgramStatus(
-    ProgramStatus.BOOTSTRAP_QUERY_RUNNING_FINISHED
+  const service = interpret(
+    // eslint-disable-next-line new-cap
+    Machine(developConfig, {
+      services: {
+        initializeDataLayer: dataLayerMachine,
+        initialize,
+      },
+      actions: {
+        assignStoreAndWorkerPool: assign<IBuildContext, DoneEventObject>(
+          (_context, event) => {
+            const { store, workerPool } = event.data
+            return {
+              store,
+              workerPool,
+            }
+          }
+        ),
+        assignDataLayer: assign<IBuildContext, DoneEventObject>(
+          (_, { data }): DataLayerResult => data
+        ),
+      },
+    }).withContext({ program, parentSpan: bootstrapSpan, app })
   )
-
-  await db.saveState()
-
-  await waitUntilAllJobsComplete()
-  requiresWriter.startListener()
-  db.startAutosave()
-  queryUtil.startListeningToDevelopQueue({
-    graphqlTracing: program.graphqlTracing,
+  service.onTransition(state => {
+    reporter.verbose(`transition to ${JSON.stringify(state.value)}`)
   })
-  queryWatcher.startWatchDeletePage()
-  const app = express()
```

Comment:
Is it relevant that express used to start _after_ the bootstrap phase? Or is it actually faster this way due to using async cycles / parallel?

Reply:
It's created, but not started. It shouldn't make any difference at the moment, but in the full state machine we spawn the dev server in a service, so want the app in context